### PR TITLE
add some features to virbos-setup

### PR DESCRIPTION
**Pull request by abrik1**
this build of virbos-setup can:-

- detect swap,/home partition
- format partitions
- mount partitions

### CODE REVIEW

**Comment by jtbx** [comment]
Hey, I recently merged #3 which adds setup_system, which basically installs the whole system (I haven't finished it yet though). Could you remove the merge conflicts and the parts which format/mount the partitions?

File: virbos-setup [hunk 1]
```diff
@@ -214,6 +214,11 @@ getlayout() {
 			ask_nodefault "What partition would you like to use for swap space?" "$partitions"
 			partition_swap="$answer"
 		fi
+		ask_yn "Do you want to use a different partition for /home?"n
```
Comment (jtbx) [inline virbos-setup]: There should be a space between `n` and the first argument, otherwise `n` gets counted as part of the first argument

File: virbos-setup [hunk 2]
```diff
@@ -214,6 +214,11 @@ getlayout() {
...
+		if [ "$answer" = 'y' ];then
+			ask_nodefault "What partition would you like to use for swap space?" "$partitions"
```
Comment (jtbx) [inline virbos-setup]: This should say something like "What partition would you like to use in place of /home?"

**Review by jtbx — CHANGES_REQUESTED** [review]

**Comment by abrik1** [comment]
hmm... well i will fix it

File: virbos-setup (hunk 2 above)
Comment (abrik1) [reply virbos-setup]: resolved

File: virbos-setup [hunk 3]
```diff
@@ -209,15 +209,21 @@ getlayout() {
...
 			fi
 		done
-		ask_nodefault "What partition would you like to use as the boot partition?" "$partitions"
-		partition_boot=/dev/"$answer"
 		ask_yn "Do you want to install a bootloader?" y
-		[ "$answer" = y ] && use_grub=1
+		if [ "$answer" = 'y' ]; then
+			ask_nodefault "What partition would you like to use for the bootloader?" "$partitions"
+			partition_boot="$answer"
+		fi
```
Comment (jtbx) [inline virbos-setup]: Please revert this change as there must be a partition for the kernel executable. It doesn't need to contain the bootloader, either.

**Review by jtbx — CHANGES_REQUESTED** [review]

File: virbos-setup (hunk 3 above)
Comment (jtbx) [reply virbos-setup]: Oh sorry, that change is from before #3. I'll just replace those lines with what I added in my pr, sorry :grimacing:
